### PR TITLE
[runtime] Use MonoError for appdomain.c functions

### PR DESCRIPTION
**Pull request by ludovic-henry**

### CODE REVIEW

**Comment by ludovic-henry** [comment]
@lambdageek thank you for the review! I updated the function.

File: mono/metadata/appdomain.c [hunk 1]
```diff
@@ -223,14 +223,22 @@ create_domain_objects (MonoDomain *domain)
  *
  */
 void
-mono_runtime_init (MonoDomain *domain, MonoThreadStartCB start_cb,
-		   MonoThreadAttachCB attach_cb)
+mono_runtime_init (MonoDomain *domain, MonoThreadStartCB start_cb, MonoThreadAttachCB attach_cb)
 {
 	MonoError error;
+	mono_runtime_init_checked (domain, start_cb, attach_cb, &error);
+	mono_error_raise_exception (&error);
```
Comment (kumpera) [inline mono/metadata/appdomain.c]: This is a runtime API, it can't raise.
Just discard the error and ban it.

File: mono/metadata/appdomain.c [hunk 2]
```diff
@@ -334,12 +345,23 @@ void
 mono_context_init (MonoDomain *domain)
 {
 	MonoError error;
+	mono_context_init_checked (domain, &error);
+	mono_error_raise_exception (&error);
```
Comment (kumpera) [inline mono/metadata/appdomain.c]: Same thing. This is an API function. Swallow the error and ban it.

File: mono/metadata/appdomain.c [hunk 3]
```diff
@@ -407,11 +429,11 @@ mono_domain_create_appdomain (char *friendly_name, char *configuration_file)
 
 	klass = mono_class_from_name (mono_defaults.corlib, "System", "AppDomainSetup");
 	setup = (MonoAppDomainSetup *) mono_object_new_checked (mono_domain_get (), klass, &error);
-	mono_error_raise_exception (&error); /* FIXME don't raise here */
+	mono_error_raise_exception (&error);
```
Comment (kumpera) [inline mono/metadata/appdomain.c]: I don't quite get why you're removing the FIXME here.

File: mono/metadata/appdomain.c [hunk 4]
```diff
@@ -566,10 +604,21 @@ MonoReflectionAssembly *
 mono_domain_try_type_resolve (MonoDomain *domain, char *name, MonoObject *tb)
 {
 	MonoError error;
+	MonoReflectionAssembly *ret = mono_domain_try_type_resolve_checked (domain, name, tb, &error);
+	mono_error_raise_exception (&error);
```
Comment (kumpera) [inline mono/metadata/appdomain.c]: This is an API function, same thing as above.

**Comment by kumpera** [comment]
And looks like @lambdageek did the exact same review. Nice!

**Comment by ludovic-henry** [comment]
@kumpera @lambdageek I updated according to your reviews, thank you!

**Pull request merged**